src/test/java/com/google/sps/servlets/UserFoldersServletTest.java

```diff
@@ -0,0 +1,180 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.After;
+
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.sps.data.Folder;
+import java.util.List; 
+import java.util.ArrayList;
+
+@RunWith(JUnit4.class)
+public final class UserFoldersServletTest {
+
+  private static final String USER_KIND = "User";
+  private static final String USER_ID = "testId";
+  private static final String FOLDER_A = "First Folder";
+  private static final String FOLDER_B = "Second Folder";
+  private static final String FOLDER_LANGUAGE = "en";
+
+  private final LocalServiceTestHelper helper = 
+    	new LocalServiceTestHelper(
+      	new LocalDatastoreServiceTestConfig()
+          .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+      	new LocalUserServiceTestConfig())
+      	.setEnvIsAdmin(true).setEnvIsLoggedIn(true)
+      	.setEnvEmail("test@gmail.com").setEnvAuthDomain("gmail.com");
+
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private UserFoldersServlet servlet;
+  private DatastoreService datastore;
+    
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    servlet = new UserFoldersServlet();
+    mockRequest = mock(HttpServletRequest.class);
+    mockResponse = mock(HttpServletResponse.class);
+    
+    // Set up a fake HTTP response 
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    helper.tearDown();
+  }
+  
+  @Test
+  public void queryUserFolders() throws Exception {
+    // Generate testing folders to store in datastore
+    Folder folderA = new Folder(FOLDER_A, FOLDER_LANGUAGE);
+    Folder folderB = new Folder(FOLDER_B, FOLDER_LANGUAGE);
+    
+    // Generate testing user entity to query by email
+	  // And generate a userKey to set as the parent of the testing folders
```

Comment:
I don't know why there is spacing on github. On my IDE, it seems fine...

Reply:
It's because there is a tab instead of a space. You have your tab-size on your IDE set to 2 spaces; on GitHub they display it as 4.